gc/base/standard/ConcurrentSweepScheme.cpp

```diff
@@ -1530,6 +1529,117 @@ omrtty_printf("]");
 }
 
 /**
+ * Due to heap resize at the end of global gc(after sweepForMinimumSize),
+ * some concurrent sweepStates (SweepChunk, heapSizeToConnect, heapSizeConnected, ApproximateFree...) need to be updated
+ * without this updating, the free memory, which is shifted from LOA to SOA, would be lost until the next global gc and memory pool reportings
+ * in verbose gc and mxbean are wrong.
+ * TODO:
+ */
+void
+MM_ConcurrentSweepScheme::updateSweepStates(MM_EnvironmentBase* env, uintptr_t resizeType)
```

Comment:
In general, there are 32-2048 chunks(depends on gc threads number) for whole heap, the chunks in LOA are average 10-20(max 200), so this function should be minor, also we only do once every global gc (plus in case concurrent sweep + LOA resized).